src/main/java/com/dreamhorizon/enchantments/EnchantmentsHandler.java

```diff
@@ -35,25 +39,27 @@
 
 public class EnchantmentsHandler {
     private static final EnchantmentsHandler instance = new EnchantmentsHandler();
-    private static final Pattern ENCHANTMENT_LORE_PATTERN = Pattern.compile("(.*) (X|IV|V?I{0,3})");
+    private static final Pattern ENCHANTMENT_LORE_PATTERN = Pattern.compile("(.*) (\\d+|X|IV|V?I{0,3})");
     
     private EnchantmentsHandler() {
         DHEnchantments.all.forEach(this::register);
     }
     
-    public void register(@NotNull Enchantment enchantment) {
+    private void register(@NotNull Enchantment enchantment) {
         try {
+            // Unregister it before registering it again. This is for reloading to work.
+            unregister(enchantment);
             Field acceptingNew = Enchantment.class.getDeclaredField("acceptingNew");
             acceptingNew.setAccessible(true);
             acceptingNew.set(null, true);
             Enchantment.registerEnchantment(enchantment);
-        } catch (Exception ex) {
-            ex.printStackTrace();
+        } catch (IllegalAccessException | NoSuchFieldException e) {
+            e.printStackTrace();
         }
     }
     
-    @SuppressWarnings("unchecked")
-    public void unregister(@NotNull Enchantment enchantment) {
+    @SuppressWarnings({"unchecked", "deprecation"})
```

Comment:
What are the deprecation warnings? Why are we suppressing them? Will these stop working at some point? What is the updated way of doing this if any?

Reply:
The updated method is getKey, which if you look inside the method is used. the thing is bukkit still keeps a name map, although it is deprecated it is still being used, which is why I need to remove it from there aswell, hence I use the deprecated Enchantment.getName() method to remove it from the deprecated byName static map in org.bukkit.enchantments.Enchantment.

Reply:
The unchecked warning I presume you know is just map/array casting, which can be done in a checked manner but requires 2 for loops, 1 array init and 2 if clauses, this is why it's usually better to just cast and catch a ClassCastException , which in our case can't even be spit out since the map we get is always of that type.

Reply:
Okay, sounds good. Thanks for the clarification